# Toggle loop closures on or off. Setting this to off will increase run-time
# speed by a small fraction.
check_for_loop_closures: true

# Default parameters for the ISAM2 data structure.
relinearize_skip: 1
relinearize_threshold: 0.01

# Amount of translational distance the sensor must travel before adding a new
# pose to the pose graph. This helps keep the graph sparse so that loop closures
# are fast.
translation_threshold: 0.5

# When searching through old poses for loop closures, only consider old poses
# that are within this distance of the current pose.
proximity_threshold: 1.5

# To compute a loop closure we perform ICP between the current scan and laser
# scans captured from nearby poses. In order to be considered a loop closure,
# the ICP "fitness score" must be less than this number.
max_tolerable_fitness: 0.15

# Don't attempt loop closures with poses in the graph that were collected within
# the last 'skip_recent_poses' poses.
skip_recent_poses: 40

# If a loop has recently been closed, don't close a new one for at least this
# many poses (i.e. the sensor would have to move translation_threshold *
# poses_before_reclosing meters in order to close a new loop).
poses_before_reclosing: 40
